checker/jdk/index/src/java/lang/String.java

```diff
@@ -162,7 +162,7 @@ public String() {
      * @param  original
      *         A {@code String}
      */
-    public @PolyMinLen @PolySameLen String(@PolyMinLen @PolySameLen String original) {
+    public /*@ PolyMinLen*/ @PolySameLen String(/*@ PolyMinLen*/ @PolySameLen String original) {
```

Comment:
Try this:
```
void test() {
        @MinLen(4) String s = String.valueOf(true);
    }
```
A annotation must have `@Target(TYPE_USE)` for it to be read from comments.   `@IndexFor` doesn't have that so it is ignored if it is in comments (with or without the spaces).  That's why you didn't run into this problem before.

But yes, you should probably fix that.  And IndexFor (and related annotations) should be have `@Target({TYPE_USE, TYPE_PARAMETER})`